qunicorn_core/core/mapper/provider_mapper.py

```diff
@@ -13,13 +13,30 @@
 # limitations under the License.
 
 from qunicorn_core.api.api_models import ProviderDto
+from qunicorn_core.api.api_models.provider_assembler_language_dtos import ProviderAssemblerLanguageDto
+from qunicorn_core.core.mapper import provider_assembler_language_mapper
 from qunicorn_core.core.mapper.general_mapper import map_from_to
+from qunicorn_core.db.models.provider_assembler_language import ProviderAssemblerLanguageDataclass
 from qunicorn_core.db.models.provider import ProviderDataclass
 
 
 def dto_to_dataclass(provider_dto: ProviderDto) -> ProviderDataclass:
-    return map_from_to(provider_dto, ProviderDataclass)
+    supported_languages: list[ProviderAssemblerLanguageDataclass] = []
+    for language in provider_dto.supported_languages:
+        supported_languages.append(provider_assembler_language_mapper.dto_to_dataclass(language))
```

Comment:
mach das hier mal auch mit dieser kurz schreibweise languages = [provider_assembler_language_mapper.dto_to_dataclass(language) for language in provider_dto.supported_languages]

Reply:
habs gemacht aber ist so halt viel unübersichtlicher